### PR TITLE
feat: integration interface libs docs

### PR DESCRIPTION
**Pull request by markuczy**

### CODE REVIEW

File: docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+==== Shell UI router mechanism (Module)
+When dealing with modules, Shell UI is responsible for loading them. As explained in the xref:architecture-overview/shell.adoc[Shell overview], Shell UI first constructs workspace and then handles routing and loading of remotes. Workspace configuration contains all valid routes to Applications registered for a given workspace, each route associated with a Microfrontend's module.
+
+Prior to any routing events, Shell UI initializes router to reflect workspace configuration. Based on the data from backend services about the current workspace, Shell UI creates routing rules to use during the runtime.
+
+Let's take a workspace using user management Application as an example. Each Application in workspace configuration has to have a path defined. This path is used for routing and accessing remote entry file of Application's Microfrontend. Assuming the path is `/user`, Shell UI will create a routing rule specifying that whenever url of the page changes to one starting with `/user` it has to load the Microfrontend's module associated with user management Application.
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc:27]: whenever the url ...

File: docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+After routing rules are successfully created, each registered Application should have a Route registered in Shell UI's router. Each Route has a **loadChildren method, which is called whenever route gets activated** and other properties used for route matching.
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc:29]: whenever the route

Comment (anninowak) [reply docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc:29]: What do you mean with: "and other properties used for route matching."?

File: docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+The **loadChildren method is responsible for loading the remote module into the Shell UI's memory using module-federation**. When remote module is loaded successfully, it is used by the router to display module's content a page content pane.
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc:31]: when _the_ remote module is loaded ...

Comment (anninowak) [reply docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc:31]: it is used by the router to display module's content _as_ a page content pane.

File: docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+==== Slot Component mechanism (Remote Component)
+Remote Component loading is handled by Slot Component. Overall mechanism of Remote Component loading was explained in the xref:architecture-overview/remoteComponents.adoc[Remote Components documentation]. In the context of content loading its important to note the following aspects:
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc:34]: by _the_ Slot Component

Comment (anninowak) [reply docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc:34]: it is / it's

File: docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+Web Components are used in OneCX context for dynamic content creation. OneCX platform uses Web Components technologies to bound UI components to html elements. OneCX platform will use information from Application Store to determine what html elements have to be rendered to allow certain modules and Remote Components to bound to page content.
+
+With webcomponents method OneCX platform is only loading the code required for Microfrontend's content into the Shell UI memory and creating a "host" element for Remote Components or modules to attach into using Web Components. Microfrontend's exposed content is partially responsible for the content creation. Microfrontends have to adapt exposed modules and components so they are using Web Components Custom Elements.
+
+=== Custom Elements
+Web Components technology called Custom Elements is set of tools used to achive the bounding between components and html elements. As mentioned, OneCX platform will create html elements specified via configuration whenever a module or Remote Component has to be displayed on a page. There are 2 steps that OneCX platform takes when dealing with content loading with webcomponent expose method:
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc:16]: Custom Elements are a part of the WebComponents technology. It is a set of tools ....

File: docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+. Load expose content via module-federation into memory
+. Create html element on a page to which expose content should be bounded
+
+In the xref:implementation-details/mfe-content-loading/index.adoc#_module_federation_loading_explained[Microfrontend content loading] it was mentioned that when exposing Microfrontend content, files specified per content should take care of all setup needed to display it. With webcomponents method, those files need to contain source code, which will register Custom Element in the Custom Elements Registry. Without getting into the details of Web Components, a file used for content loading needs to use Web Components to assign some component to the html element that OneCX platform will render. This way when OneCX platform creates this html element on a page, Custom Elements Registry will use the assigned component and display it on a page.
+
+=== Webcomponent method requirements
+
+To correctly use webcomponent method 2 requirements need to be fullfield:
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc:25]: To correctly use webcomponent method, 2 requirements need to be fullfield:

File: docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+NOTE: Database configuration is easy to set up using the Application Store Application.
+
+IMPORTANT: To ensure Microfrontend compliance with Web Components for Angular modules and Remote Components it is recommended to use **@onecx/angular-webcomponents** library.
+
+// TODO: Link to guide of webcomponents expose method 
+== Expose Microfrontend's content as Web Components
+OneCX platform offers different ways of xref:architecture-overview/mfe.adoc[exposing Microfrontend's content]. One of them is **webcomponent method**.
+
+When exposing content with webcomponents method, it is required to ensure the correct setup. For exposing Angular content there are functionalities provided via `@onecx/angular-webcomponents` library. For detailed guide on how to expose Microfrontend's content via webcomponents method, please refer here.
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc:45]: For _a_ detailed guide

File: docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+
+**WebComponentLoaderModule** is a simple Angular module with a Router set up to always display **WebComponentLoaderComponent**. This component is responsible for rendering the html element required to display the module with Web Components Custom Elements. Whenever Shell UI's router detects change of the Microfrontend (another path is matched then the current one), it calls **loadChildren method** to load desired module and returns **WebComponentLoaderModule**. The module then creates and initializes **WebComponentLoaderComponent**, which after initializations creates the html element. Then Web Components using Custom Elements technology bounds the Microfrontend's module to the created element.
+
+For detailed guide on how to expose Microfrontend's module via webcomponents method, please refer here.
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc:73]: For _a_ detailed guide

File: docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+
+// TODO: Add link to example of migrating angular remote components to integrate with OneCX
+=== Remote Component behavior
+Slot Component is responsible for displaying Microfrontend's Remote Components during Slot Component initialization. With webcomponents method Slot Component will adapt the steps to:
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc:77]: With the webcomponents method, Slot Component will adapt the steps to

File: docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+
+. load desired remote content via module-federation (no change)
+. create html element directly inside a container (instead of creating Microfrontend's Remote Component instance)
+
+The Slot Component content is based on https://angular.dev/api/core/ViewContainerRef[Angular ViewContainerRef] to dynamically create content inside the Slot Component. With webcomponents method, the desired Microfrontend's Remote Component is still being loaded using module-federation. During that loading, it should register a component in Custom Elements Registry as mention in previous section. For OneCX platform it means that it no longer should create the Microfrontend's component directly, but ensure that the html code has correct element in order to display the component.
+
+The difference for webcomponent method is that from OneCX platform perspective there is no component to be created, but a "host" element needs to be created instead to allow Web Components Custom Elements to display desired content.
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc:84]: The difference with the web component method is that, from the OneCX platform’s perspective, no component needs to be created. Instead, a ‘host’ element must be created to allow Web Components’ custom elements to display the desired content.

File: docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+
+For detailed guide on how to expose Microfrontend's Remote Components via webcomponents method, please refer here.
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc:86]: For a detailed guide

File: docs/modules/general/pages/angular-integration-interface.adoc [hunk 12]
```diff
@@ -0,0 +1,298 @@
+= OneCX angular-integration-interface library
+
+== Overview
+`@onecx/angular-integration-interface` library is a set of tools aiming to ease the development of Angular applications integrated with OneCX by utilizing and extending functionalities of xref:integration-interface.adoc[@onecx/integration-interface]. This library offers Injectable services exposing set of functionalities useful for integrating Microfrontends with OneCX without utilizing Topics directly.
```
Comment (anninowak) [inline docs/modules/general/pages/angular-integration-interface.adoc:4]: [....] with OneCX without directly utilizing Topics.

File: docs/modules/general/pages/angular-integration-interface.adoc [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+|===
+|Property name |Type | Description
+
+|`remoteComponents$` | `RemoteComponentsTopic` | RemoteComponentsTopic instance.
+|===
+
+[#theme_service]
+=== Theme Service
+Usage::
+*Change page display style by applying Themes*.
```
Comment (anninowak) [inline docs/modules/general/pages/angular-integration-interface.adoc:200]: Change the page display style by applying Themes

**Review by anninowak — CHANGES_REQUESTED** [review]
Hi Martin,
I have added a few comments to your documentation. 
Technically, it is described very well and in detail.

File: docs/modules/general/pages/angular-integration-interface.adoc [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+== Overview
+`@onecx/angular-integration-interface` library is a set of tools aiming to ease the development of Angular applications integrated with OneCX by utilizing and extending functionalities of xref:integration-interface.adoc[@onecx/integration-interface]. This library offers Injectable services exposing set of functionalities useful for integrating Microfrontends with OneCX without utilizing Topics directly.
+
+== Services
+
+=== App Config Service
+Usage::
+*Initialize and access configuration variables for a specific Microfrontend*. 
++
+It is similar to xref:angular-integration-interface.adoc#configuration_service[ConfigurationService], however *App Config Service* should be used for a Microfrontend configuration management. If it is desired to share the configuration variable with many xref:angular-integration-interface.adoc#configuration_service[ConfigurationService] and provide variables via Shell Application.
```
[the remark this answers is not in the document]
Comment (markuczy) [reply docs/modules/general/pages/angular-integration-interface.adoc:12]: Something needs to be desired and without **it** in the phrase seems to me like there is no reference. Another way could be:
If sharing the configuration variable with multiple Microfrontends is desired, please use ...

File: docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+* **content type**
+** Module
+** Remote Component
+* **expose method**
+** Angular
+** Webcomponent
+
+All of them are dependent on the Application Microfrontend's content that is meant to be integrated with OneCX platform. Please, make sure to properly configure each integrated Application in Application Store.
+
+IMPORTANT: **It is NOT recommended** to use Angular method. Doing so restraints the platform from using many frameworks with different versions, which means that all Microfrontends would need to use the same technology and version.
```
[the remark this answers is not in the document]
Comment (markuczy) [reply docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc]: Throughout the docs I am distinguishing 2 methods of Microfrontend content exposal. One is Webcomponent expose method and the other one is Angular expose method.
Both of them were first introduced in architecture-overview/mfe.adoc file.

File: docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc (hunk 2 above)
Comment (markuczy) [reply docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc]: The intention here is to highlight that each route has the load children method, but it also has additional properties that are used by the shell during routing. I refactored it a bit so its more clear now maybe? Let me know what you think and maybe its better to remove the properties reference

Comment (anninowak) [reply docs/modules/general/pages/implementation-details/mfe-content-loading/index.adoc]: What do you think of the following?

Each Route has a **loadChildren method, which is called whenever route gets activated**, but it also has additional properties that are used by the shell during routing.

File: docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc [hunk 16]
```diff
@@ -0,0 +1,205 @@
...
+
+IMPORTANT: Webcomponent expose method enables the OneCX platform to use UI Apps developed with different frameworks and versions. It is the **recommended method for all Applications**.
+
+NOTE: Web Components technologies are used within OneCX platform, however OneCX offers its own functionalities that should be used to expose Microfrontend's content using the Webcomponent expose method. Those functionalities are based on Web Components technologies, so there is no need to get familiar with them. Nevertheless, more resources about Web Components can be found https://developer.mozilla.org/en-US/docs/Web/API/Web_components[here]
+
+== Web Components preview
+Web Components are used in OneCX context for dynamic content creation. OneCX platform uses Web Components technologies to bound UI components to html elements. OneCX platform will use information from Application Store to determine what html elements have to be rendered to allow certain modules and Remote Components to bound to page content.
```
Comment (BenjaminPabst) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc:11]:
```suggestion
Web Components are used in OneCX context for dynamic content creation. OneCX platform uses Web Components technologies to bind UI components to html elements. The OneCX platform will use information from the Application Store to determine what html elements have to be rendered to allow certain modules and Remote Components to be bound to page content.
```

File: docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc [hunk 17]
```diff
@@ -0,0 +1,205 @@
...
+NOTE: Database configuration is easy to set up using the Application Store Application.
+
+// TODO: Add link to bootstrapping
+IMPORTANT: To ensure Microfrontend compliance with Web Components for Angular modules and Remote Components it is recommended to use **<<onecx/angular-webcomponents, @onecx/angular-webcomponents>>** library.
+
+// TODO: Link to guide of Webcomponent expose method 
+== Expose Microfrontend's content as Web Components
+OneCX platform offers different ways of xref:architecture-overview/mfe.adoc[exposing Microfrontend's content]. One of them is **Webcomponent method**.
+
+// TODO: Link to guide of Webcomponent expose method
```
Comment (BenjaminPabst) [inline docs/modules/general/pages/implementation-details/mfe-content-loading/webcomponents.adoc:46]: duplicated todo?

**Pull request closed without merging**